File: counting/counting.py
import asyncio

async def count(numbers):
    for num in numbers:
        print(num)
        await asyncio.sleep(0.1)  # 100ms delay

async def main():
    print("Counting from 1 to 10 in python")
    numbers = range(1, 11)
    await count(numbers)

asyncio.run(main())

File: counting/counting.ts
console.log("Counting from 1 to 10 in TypeScript using async/await:");

const nums = Array(10).fill(null).map((_, i) => i + 1);
const delay = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));

const logNumbers = async (numbers:number[]) => {
    for (const number of numbers) {
        await delay(100);
        console.log(number);
    }
}

logNumbers(nums);
